modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java

```diff
@@ -149,16 +149,18 @@ public void setFlowConstruct(FlowConstruct flowConstruct)
 
     private MetadataContext getMetadataContext() throws MetadataResolvingException
     {
-        //TODO MULE-9530: Improve Config retrieval for Metadata resolution
-        if (!StringUtils.isBlank(configurationProviderName) &&
-            muleContext.getRegistry().get(configurationProviderName) instanceof DynamicConfigurationProvider)
+        MuleEvent fakeEvent = getInitialiserEvent(muleContext);
+        ConfigurationInstance<Object> configuration = getConfiguration(fakeEvent);
```

Comment:
The provider is only acquired to verify that it is not of Dynamic type. If we first tried to fetch the provider, and then obtain the configuration from it (like we did before), we could face the case where the provider is not present, forcing us to trigger the implicit configuration creation, and then again fetch the provider to verify if the one created for the implicit configuration is dynamic or not.

I think is cleaner just to get the configuration, let the manager do its magic if the config has to be created implicitly, and then verify if the registered provider is dynamic or not.